### ISSUE TITLE
NC client does not react to ACL permission changes

### ISSUE THREAD

**Issue by TheRealTripleB** [nextcloud/desktop#3672]
### Expected behaviour
Existing ACL Group Folder 'Projects' with multiple directorys (Project1, Project2, ...) with full access rights for group 'X' containing 'User1' and 'User2'.
Both users have fully synced the hierarchy and all files with their local drive using NC client.
Now I create a new access rule for 'User2' for subfolder 'Project1': read access -> 'deny'
Expected: Subfolder 'Project1' is deleted from the local drive of 'User2' by the NC client

### Actual behaviour
NC client does not react on the change of permission.
Only after 'User2' restarts the NC client the 'Project1' disappears, but only from the list of folders for selective sync, however the content of the folder is still not deleted on the local drive. I noticed that the subfolder is only deleted when a file/folder is changed in this folder by another user. In the log you can see that the Nextcloud client then gets the request to delete the directory (Propagate Local Remove). Without a change in the subfolder this does not happen and the folder exists forever on the computer.

### Steps to reproduce
1. Remove read access for a subfolder for one person of the group
2. Restart the nc client

The problem is already described in Issue #1353. Before this was closed I thought the bug was fixed until I looked at the issue again now because I want to use this setup like this.

### Client configuration
Mac Client Version: 3.3.0

Operating system: MacOS BigSur

OS language: German (DE)

### Server configuration
Nextcloud version: 21.0.1

### Logs
1. Client logfile: 
I don't have an anonymized log file at the moment, but I can provide it with some time if needed.

**Comment by ChrisgammaDE** [comment]
Similar thing for me to:
We manage group folder access via groups.

When we change the group. A user can see the folder instantly after forcing a synchronisation.
But, when I add the user to a group, that allows them to create a file in this folder, the client doesn't notice this.

Steps to reproduce:
1. Create two groups. Let's call them "reader" and "editor".
2. Create a group folder and set the permissions so that reader can only read, and editor can do everything else. (Might as well set yourself admin)
3. Create some example folders and files in the group folder for testing
4. Add user to group "reader". In the Client force a synchrnoisation.
5. User is able to browse all files, but when creating a file, the client (rightfully) doesn't sync it to Nextcloud.
6. Add user to group "editor". It now is in two groups. In the Client force a synchrnoisation.
7. User is still able to browse all files, but but when creating a file, the client (now wrongfully) still doesn't sync it to Nextcloud.

To get the desired action for step 7 I found the following workaround:
As Nextcloud in the webbrowser does not have this issue, user can upload his file manually to the folder and the Client will synchronise it automatically. While synchronising, the client also updates the permission of the folder.
Therefore, after this "hack" the folder is synced again by the client.

As far as I can tell this sounds like a "ACL cache-busting"-thing

**Comment by TheRealTripleB** [comment]
I did some more research on the problem over the weekend and set up a new Nextcloud 22 instance.I can also reproduce this problem, but I also believe that this is more of a client problem, since it works in the browser. Therefore I have also tried the Owncloud client with the Nextcloud instance and could determine the same behavior.

It would be nice if this bug is fixed, we would like to use the extended permissions in our company and put more on nextcloud, but unfortunately this is not currently possible due to this bug. 

Is there any further information needed to reproduce the problem and fix the bug?

**Comment by Rello** [comment]
Hello,

this ticket is pretty old and the client had several fundamental redesigns.
If the issue still exists, please create a new ticket with current information.

Thank you
